akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -945,7 +948,30 @@ private[akka] class ShardRegion(
     if (shardBuffers.contains(shardId)) {
       val buf = shardBuffers.getOrEmpty(shardId)
       log.debug("{}: Deliver [{}] buffered messages for shard [{}]", typeName, buf.size, shardId)
-      buf.foreach { case (msg, snd) => receiver.tell(msg, snd) }
+
+      buf.foreach {
+        case (msg, snd) =>
+          msg match {
+            case msg @ RestartShard(_) =>
```

Comment:
This is where we need to decide either where to re-route or other. As a draft I'm just checking and logging debug, unless it seems the shard is local to the region.

Reply:
`RestartShard` is sent from one place, in receiveTerminated via scheduleOnce, when the (local) Shard is terminated (and rememberEntities enabled).

When RestartShard is later received it's buffered because the shard location is unknown. Request location from coordinator with GetShardHome.

Then receiving ShardHome and `if (shardRegionRef != self)` will call this `deliverBufferedMessages` with a `receiver` which is a ShardRegion at another node.

I think we can just ignore/drop the buffered RestartShard message here if `receiver != self`.

If the shard has been relocated it should be the responsibility of that other ShardRegion (and/or the coordinator) to make sure the shard is started.